File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.java
class Solution {
    public boolean validPalindrome(String s) {
        int left=0;
        int right=s.length()-1;
        while(left<right){
            if(s.charAt(left)==s.charAt(right)){
                left++;
                right--;
            }else{
                return isPalin(s,left+1,right) || isPalin(s,left,right-1);
            }
        }
        return true;
    }
    public boolean isPalin(String s,int left,int right){
        while(left<right){
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/rotate-matrix/submission-0.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n=len(matrix)
        for i in range(n):
            for j in range(i+1,n):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
            
        for row in range(n):
            left=0
            right=n-1
            while left<right:
                matrix[row][left],matrix[row][right]=matrix[row][right],matrix[row][left]
                left+=1
                right-=1
              